pkg/clients/clients.go

```diff
@@ -44,32 +43,23 @@ func init() {
 }
 
 // GetKubeClientSetOrDie Add a native non-caching client for advanced CRUD operations
-func GetKubeClientSetOrDie() kubernetes.Clientset {
-
-	clientSet, err := kubernetes.NewForConfig(RestConfig)
-	exit.OnError(err)
-	return *clientSet
+func GetKubeClientSetOrDie() (*kubernetes.Clientset, error) {
+	return kubernetes.NewForConfig(RestConfig)
 }
 
 // GetConfigClientOrDie Add a configv1 client to the reconciler
-func GetConfigClientOrDie() clientconfigv1.ConfigV1Client {
-
-	client, err := clientconfigv1.NewForConfig(RestConfig)
-	exit.OnError(err)
-	return *client
+func GetConfigClientOrDie() (*clientconfigv1.ConfigV1Client, error) {
+	return clientconfigv1.NewForConfig(RestConfig)
 }
 
-func GetCachedDiscoveryClientOrDie() discovery.CachedDiscoveryInterface {
-
-	client, err := config.ToDiscoveryClient()
-	exit.OnError(err)
-	return client
+func GetCachedDiscoveryClientOrDie() (discovery.CachedDiscoveryInterface, error) {
+	return config.ToDiscoveryClient()
```

Comment:
Also changes the behaviour of the function. Are those used a lot? If not, I'd be in favour of removing them entirely.